### PR TITLE
Code for Sorting Array of 0s 1's and 2's in C++

### LINKED ISSUE #1161: Adding Sort an array of 0s, 1s and 2s in C++

### PR DESCRIPTION
**Pull request by HariniJeyaraman**
Fixes #1161 

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
- Implemented Sort in Array containing only 0's 1's and 2's in C++
- 

#### Languages Used:
-  C++
- 

#### Files Added:
-  Sort_Array_zero_one_two.cpp 
- 

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

### CODE REVIEW

**Comment by HariniJeyaraman** [comment]
@jainaman224 @SakshiShreya please review this issue. Please let me know if I've made any mistake in the pull request/contributing to the issue (I am a beginner in contributing to open source projects)

**Comment by jainaman224** [comment]
Complete PR template.

File: Sort_Array_zero_one_two/segregate_zero_one_two.cpp [hunk 1]
```diff
@@ -0,0 +1,51 @@
+/*Given an array A[] consisting 0s, 1s and 2s. The task is to write a function that sorts the given array. 
+The function puts all 0s first, then all 1s and all 2s in last.*/
+
+#include<stdlib.h>
+#include<iostream>
+using namespace std;
+
+void sort_zero_one_two(int a[],int n)
+{
+    int zero=0; //counter to store number of 0's
```
Comment (singh-shreya6) [inline Sort_Array_zero_one_two/segregate_zero_one_two.cpp]: Add space around operators across the file.

File: Sort_Array_zero_one_two/segregate_zero_one_two.cpp [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+        cout<<a[i]<<" ";
+    cout<<"\n";
+    return 0;
+}
+
+/* Sample Input - Output
+Enter the number of elements of array: 7                                                                                                                                                                                                                  
+Enter the array elements: 1 0 2 1 1 0 2                                                                                                                                                                                                             
+Original array : 1 0 2 1 1 0 2 
+Sorted array   : 0 0 1 1 1 2 2  */
```
Comment (singh-shreya6) [inline Sort_Array_zero_one_two/segregate_zero_one_two.cpp]: add a blank line at end of file.

File: Sort_Array_zero_one_two/segregate_zero_one_two.cpp [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    for(int i=zero;i<zero+one;i++)
+        a[i]=1;
+    for(int i=one+zero;i<n;i++) //filling rest of the array with 2
+        a[i]=2;
+}
+
+int main()
+{
+    int n;
+    cout<<"Enter the number of elements of array: ";
```
Comment (singh-shreya6) [inline Sort_Array_zero_one_two/segregate_zero_one_two.cpp]: Spaces around operators.

Comment (jainaman224) [reply Sort_Array_zero_one_two/segregate_zero_one_two.cpp]: Not resolved yet.

File: Sort_Array_zero_one_two/segregate_zero_one_two.cpp [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    for(int i=0;i<n;i++)
+        cin>>a[i];
+    cout<<"Original array : ";
+    for(int i=0;i<n;i++)
+        cout<<a[i]<<" ";
+    sort_zero_one_two(a,n); //calling the sort function
+    cout<<"\nSorted array   : ";
+    for(int i = 0;i < n;i++)
+        cout<<a[i]<<" ";
+    cout<<"\n";
```
Comment (jainaman224) [inline Sort_Array_zero_one_two/segregate_zero_one_two.cpp]: Use endl

**Comment by jainaman224** [comment]
@HariniJeyaraman Are you working on it?

**Comment by HariniJeyaraman** [comment]
@jainaman224 I have made a pull request. Please review it again. I have made the changes that you proposed. Yes, I have finished working on it. Waiting to be reviewed

File: Sort_Array_zero_one_two/Sort_Array_zero_one_two.cpp [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+    for(int i = 0 ; i < n ; i++)
+        cin>>a[i];
+    cout<<"Original array : ";
+    for(int i = 0 ; i < n ; i++)
+        cout<<a[i]<<" ";
+    sort_zero_one_two(a,n); //calling the sort function
+    cout<<endl;
+    cout<<"Sorted array   : ";
+    for(int i = 0;i < n ; i++)
+        cout<< a[i] <<" ";
```
Comment (singh-shreya6) [inline Sort_Array_zero_one_two/Sort_Array_zero_one_two.cpp]: Please add space around << and >> at all places while taking input and displaying output.

File: Sort_Array_zero_one_two/Sort_Array_zero_one_two.cpp [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+    for(int i = one + zero ; i < n ; i++) //filling rest of the array with 2
+        a[i] = 2;
+}
+
+int main()
+{
+    int n;
+    cout<<"Enter the number of elements of array: ";
+    cin>>n;
+    int a[n];
```
Comment (singh-shreya6) [inline Sort_Array_zero_one_two/Sort_Array_zero_one_two.cpp]: One blank line here.

File: Sort_Array_zero_one_two/Sort_Array_zero_one_two.cpp [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+    return 0;
+}
+
+/* Sample Input - Output
+Enter the number of elements of array: 7                                                                                                                                                                                                                  
+Enter the array elements: 1 0 2 1 1 0 2                                                                                                                                                                                                             
+Original array : 1 0 2 1 1 0 2 
+Sorted array   : 0 0 1 1 1 2 2  */
+
+
```
Comment (singh-shreya6) [inline Sort_Array_zero_one_two/Sort_Array_zero_one_two.cpp]: Unnecessary extra line. One blank line is enough

File: Sort_Array_zero_one_two/Sort_Array_zero_one_two.cpp [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+int main()
+{
+    int n;
+    cout<<"Enter the number of elements of array: ";
+    cin>>n;
+    int a[n];
+    cout<<"Enter the array elements: ";
+    for(int i = 0 ; i < n ; i++)
+        cin>>a[i];
+    cout<<"Original array : ";
```
Comment (singh-shreya6) [inline Sort_Array_zero_one_two/Sort_Array_zero_one_two.cpp]: One blank line before this.

File: Sort_Array_zero_one_two/Sort_Array_zero_one_two.cpp [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+    for(int i = 0 ; i < n ; i++)
+        cout<<a[i]<<" ";
+    sort_zero_one_two(a,n); //calling the sort function
+    cout<<endl;
+    cout<<"Sorted array   : ";
```
Comment (singh-shreya6) [inline Sort_Array_zero_one_two/Sort_Array_zero_one_two.cpp]: One blank line before this for readability.

**Comment by HariniJeyaraman** [comment]
@jainaman224 @singh-shreya6 this issue has already been closed by my PR #1999 . I think you can ignore this PR. By mistake I made 2 PR's for the same issue. My apologies. PR #1999 was successfully merged for Sort Array of 0's 1's and 2's

**Pull request closed without merging**